### ISSUE TITLE
Add ability to set channel's display name when creating via CLI

### ISSUE THREAD

**Issue by XANi** [mattermost/platform#4274]
Currently every other (header/purpose/members) option of channel/group can be set except channel's Display Name, which makes any kind of automation annoying to make (as there is currently no full web API docs either and even golang client is poorly documented).

There is no option to delete/disable channel either, even tho there is "restore"

**Comment by esethna** [comment]
Thanks @XANi, our web service API documentation is a work in progress available at https://api.mattermost.com/. 

Thanks for the suggestion of adding Display name to the CLI tools for creating a channel. That makes a lot of sense. I've created a ticket to [track this in JIRA](https://mattermost.atlassian.net/browse/PLT-4441). Would you be interested in adding this functionality?

**Comment by XANi** [comment]
After evaluating it we've opted for integrating features we need via golang api. (web one is a bit under-documented for that, altho we did PoC bot that delivers cat gifs on demand via that ;))

We *basically* need synchronization between our LDAP grups and private groups in MM (on top of  existing LDAP users sync; we are currently evaluating a trial and will be getting enterprise version in a week or two) but I'm not sure if our use case is generic enough to warrant integrating with "core" of MM.

So currently I'm planning to make a daemon that gets triggered by our identity management system and sync LDAP group members with private group list, creates new ones if necessary etc. on demand (parts of it will probably land somewhere on github, eventually)
